enhancements/sig-architecture/32-extensiblescheduling/32-extensiblescheduling.md

```diff
@@ -0,0 +1,363 @@
+
+## Motivation
+
+When implementing placement resource based scheduling, we find in some cases, prioritizer needs extra data (more than the defualt value provided by `ManagedCluster`) to calculate the score of the managed cluster. For example, there is requirement to schedule based on resource monitoring data from the cluster.
+
+So we want a more extensible way to support scheduling based on customized value.
+
+### Goals
+- Design a new API(CRD) to contain a scalar value for each managed cluster.
+- Let placement prioritizer maintain the lifecycle (creating/deleting) of the CR.
+- Let placement prioritizer support rating clusters with the scalar value of the CR.
+
+### Non-Goals
+- How a customized agent update the scalar value into the CR.
+- Placement filter will not support the new API(CRD).
+
+## Proposal
+
+### User Stories
+
+#### Story 1: Users could use the data pushed from each managed cluster to select clusters.
+  - On each managed cluster, there is a customized agent monitoring the resource usage (eg. CPU ratio) of the cluster. It will calculate a scalar value and push the result to hub.
+  - As an end user, I can configure placement yaml to use this scalar value to select clusters.
+  
+#### Story 2: Users could use the metrics collected on hub to select clusters.
+  - On the hub, there is a customized agent to get metrics (eg. cluster allocatable memory) from Thanos. It will generate a scalar value for each cluster.
+  - As an end user, I can configure placement yaml to use this scalar value to select clusters.
+
+#### Story 3: Disaster recovery workload could be automatically switched to an avaliable cluster.
+  - A user has two clusters, a primary and backup cluster on which workload storage is configured so storage are synced from primary cluster to backup.
+  - As an end user, I want to deploy workloads on the primary cluster first. And when primary cluster is down, the workload should be automatically switched to the backup cluster.
+
+#### Story 4: Users could use the data provided by third party controllers to select clusters.
+  - A third party controller can comprehensively evaluate cluster on latency, region, iops etc. and rate cluster according to the SLA the cluster can provide.
+  - As an end user, I can configure placement yaml to use this scalar value to select clusters.
+
+### Risks and Mitigation
+N/A
+
+## Design Details
+
+### ManagedClusterScalar API
+ManagedClusterScalar is the new API to add, to represents a scalable value (aka score) of one managed cluster.
+```go
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:scope="Namespaced"
+// +kubebuilder:subresource:status
+
+// ManagedClusterScalar represents a scalable value (aka score) of one managed cluster.
+// Each ManagedClusterScalar only represents the scalar value of one prioritizer.
```

Comment:
one prioritizer per entry is definitely the cleanest, but it seems like we're likely to want a single controller in the managedcluster reporting back several prioritizer values.  Have we considered allowing multiple scalars reported back to amortize the cost of updates with many prioritizers from a single source?

Reply:
yes, will try to modify this part. 

It might be like letting the API contains a group of prioritizer values. So that on the managed cluster, one controller updates cpu/mem/cpuratio/memratio values into one CR, another controller updates other customized values, eg, GPU, into another CR.